oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -272,6 +303,39 @@ public LowLevelHttpResponse execute() throws IOException {
     };
   }
 
+  private MockLowLevelHttpRequest getMockRequestForMtlsConfig(String url) {
+    return new MockLowLevelHttpRequest(url) {
+      @Override
+      public LowLevelHttpResponse execute() throws IOException {
+
+        String metadataRequestHeader = getFirstHeaderValue(S2A.METADATA_FLAVOR);
+        if (!S2A.GOOGLE.equals(metadataRequestHeader)) {
+          throw new IOException("Metadata request header not found");
+        }
+
+        // Create the JSON response
+        GenericJson content = new GenericJson();
+        content.setFactory(OAuth2Utils.JSON_FACTORY);
+        if (requestStatusCode < 400) {
```

Comment:
nit: Can this be 200? Or are there other 2xx/ 3xx codes?

Reply:
Yeah it can just be 200. We expect only 200 on successful responses. Done in https://github.com/googleapis/google-auth-library-java/pull/1400/commits/20825f70a2e6e21a03a4480267f62f706f7ea35f